File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.js
function solution(n, w, num) {
    let answer = 0;
    
    const boxList = [];
    let flagBoxList = [];
    let flag = true;
    
    for (let i = 1; i <= n; i++) {
        flagBoxList.push(i);
        
        if (flagBoxList.length === w) {
            if (!flag) {
                flagBoxList.reverse();
            }
            
            boxList.push(flagBoxList);
            flag = !flag
            flagBoxList = [];
        }
    }
    
    // 마지막 줄이 w개 미만으로 남은 경우
    if (flagBoxList.length > 0) {
        const listLength = flagBoxList.length;
        
        if (flag) {
            while (flagBoxList.length < w) {
                flagBoxList.push(0);
            }
            boxList.push(flagBoxList);
        } else {
            flagBoxList.reverse();
            const finalRow = Array(w - listLength).fill(0).concat(flagBoxList);
            boxList.push(finalRow);
        }
    }

    // num의 위치 찾기
    let targetRow = 0;
    let targetCol = 0;
    
    for (let r = 0; r < boxList.length; r++) {
        const col = boxList[r].indexOf(num);
        
        if (col !== -1) {
            targetRow = r;
            targetCol = col;
            break;
        }
    }
    
    // 같은 열에 있는 상자 개수 세기
    for (let r = targetRow; r < boxList.length; r++ ) {
        if (boxList[r][targetCol] !== 0) {
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.py
def solution(n, w, num):
    answer = 0
    
    box_list = []
    flag_box_list = []
    flag = True
    for i in range(1, n+1):
        flag_box_list.append(i)
        if len(flag_box_list) == w:
            if not flag: 
                flag_box_list.reverse()

            box_list.append(flag_box_list)
            flag = not flag
            flag_box_list = []
    
    # 마지막 줄이 w개 미만으로 남았을 때 
    if flag_box_list:
        list_length = len(flag_box_list)
        if flag:
            flag_box_list.extend([0] * (w - list_length))
            box_list.append(flag_box_list)
        else:
            flag_box_list.reverse()
            final_row = [0] * (w - list_length) + flag_box_list
            box_list.append(final_row)
        
    print(box_list)
    
    # 인덱스 값 찾기
    i = j = 0
    for r, row in enumerate(box_list):
        if num in row:
            i = r
            j = row.index(num)
            break
            
    for r in range(i, len(box_list)):
        if box_list[r][j] != 0:
            answer += 1
            
    return answer